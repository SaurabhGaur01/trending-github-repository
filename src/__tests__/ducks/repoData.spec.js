import {
    setRepoData,
    clearRepoData,
    repoDataReducer,
} from '../../ducks/repoData';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('repoData Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setRepoData should match the snapshot' , () => {
            expect(setRepoData(mockObject)).toMatchSnapshot();
        });
        it('clearRepoData should match the snapshot' , () => {
            expect(clearRepoData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(repoDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(repoDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(repoDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value of the API call' , () => {
            expect(repoDataReducer('', setRepoData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearRepoData' , () => {
            expect(repoDataReducer(mockObject, clearRepoData())).toEqual([]);
        });
    });
});