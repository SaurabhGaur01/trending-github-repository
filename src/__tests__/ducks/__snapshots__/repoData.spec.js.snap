// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repoData Duck Tests Action Generator clearRepoData should match the snapshot 1`] = `
Object {
  "type": "modules/repo-app/CLEAR_REPO_DATA",
}
`;

exports[`repoData Duck Tests Action Generator setRepoData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/repo-app/SET_REPO_DATA",
}
`;
